packages/evm/contracts/payments/PaymentsBook.sol

```diff
@@ -383,6 +338,7 @@ contract PaymentsBook is ControlledRecoverable, TimeHelpers, IPaymentsBook {
         if (_token == address(0)) {
             require(msg.value == _amount, ERROR_ETH_DEPOSIT_MISMATCH);
         } else {
+            require(msg.value == 0, ERROR_ETH_DEPOSIT_MISMATCH);
```

Comment:
I don't see how it would be optimized tbh, you still need to check that `msg.value == _amount`, so you end up having three `require`s anyway. Also, it sounds weird not checking the token address in the second case.

Reply:
Discussed offline, and good catch on the second require with `msg.value == _amount`.

The thinking is that putting more requires in the ETH branch is better, because:

1. Receiving tokens will likely be more popular
2. The ETH branch is cheaper, so we "smooth out" the cost a bit between the branches.